node/silkworm/downloader/internals/db_tx_test.cpp

```diff
@@ -87,6 +87,52 @@ namespace db {
 
             REQUIRE(read_stage_progress == stage_progress);
         }
+
+        SECTION("write/read/delete canonical hash") {
+            CHECK_NOTHROW(tx.write_canonical_hash(header.number, header.hash()));
+            CHECK_NOTHROW(tx.delete_canonical_hash(header.number));
+
+            auto read_hash = tx.read_canonical_hash(header.number);
+
+            REQUIRE(read_hash == std::nullopt);
```

Comment:
In such tests where each step conditions the execution of the subsequents always prefer `REQUIRE` vs `CHECK`
The first stops the test if the condition is (not) met while the latter makes the test to continue and this may cause the following tests to produce misleading messages.
```suggestion
            REQUIRE_NOTHROW(tx.write_canonical_hash(header.number, header.hash()));
            REQUIRE_NOTHROW(tx.delete_canonical_hash(header.number));

            auto read_hash = tx.read_canonical_hash(header.number);

            REQUIRE(read_hash == std::nullopt);
```